lib/widgets/message_list.dart

```diff
@@ -70,6 +71,27 @@ class MessageListAppBarTitle extends StatelessWidget {
 
   final Narrow narrow;
 
+  Widget _buildStreamRow(ZulipStream? stream, String text) {
+    final icon = switch (stream) {
+      ZulipStream(isWebPublic: true) => ZulipIcons.globe,
+      ZulipStream(inviteOnly: true) => ZulipIcons.lock,
+      ZulipStream() => ZulipIcons.hash_sign,
+      null => null, // A null [Icon.icon] makes a blank space.
+    };
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.baseline,
+      textBaseline: TextBaseline.alphabetic,
+      children: [
+        Padding(
+          // TODO(design): The vertical alignment of the stream privacy icon is ad hoc and eyeballed.
```

Comment:
> One thing I notice (with help from "show baselines" in the DevTools), if sticking with `baseline`: the icon's baseline always seems to be its bottom edge, regardless of any vertical padding. So, adjusting the 4dp bottom padding to be big or small doesn't actually change how the icon is aligned with the text.

Huh, good catch. I must have added the padding, then switched to baseline alignment, then not checked if the padding was still needed.



> Using `CrossAxisAlignment.center` and removing the 4dp bottom padding is an improvement, to my eye; what do you think:

Right, so that's the default. Here's what that looks like for me:

![image](https://github.com/zulip/zulip-flutter/assets/28173/eac51419-3c93-47e4-918f-82a7e5b65075)

The icon sticks out much more below the text than above it. The difference from your test is presumably that I have my system font size set to small.

If I increase my system font-size setting to the default, it's better but still seems slightly below center visually. I think it's the same as your second screenshot above. Or here it is (still with center alignment) when the text starts with "f" instead of "i":
![image](https://github.com/zulip/zulip-flutter/assets/28173/b549175b-7632-4b1d-9764-cc350d84407a)
That feels low — both the top and bottom of the icon are below the top and bottom respectively of the "f".

Here's that same screen with baseline alignment:
![image](https://github.com/zulip/zulip-flutter/assets/28173/743dec92-62fb-443e-ae8e-29d6c706ff5b)
So I think baseline alignment works out better there than center, even at default font size.

(Going further up the scale, if I crank the font size up to large, I actually get the same results with either center or baseline alignment:
![image](https://github.com/zulip/zulip-flutter/assets/28173/0bf5f4fd-c265-4281-b2d9-7d7ab04b6664)
and they seem pretty OK.)